src/plugins/authenticate/index.ts

```diff
@@ -1,20 +0,0 @@
-import { authenticate } from './authenticate'
-import { beforeRequest } from './before-request'
-
-type APIClient = import('./types').APIClient
-type AuthenticatePluginState = import('./types').AuthenticatePluginState
-type Options = import('./types').Options
-
-function authenticatePlugin(client: APIClient, clientOptions: Options): void {
-  if (clientOptions.auth) return
-
-  const state: AuthenticatePluginState = {
-    client,
-    auth: undefined,
-  }
-
-  client.authenticate = authenticate.bind(null, state)
-  client.requestHook.before(beforeRequest.bind(null, state))
-}
-
-export default authenticatePlugin
```

Comment:
How is removing this plugin related to this PR?

Reply:
I don't think there is a need for authenticate, I moved it's logic to ```auth/basicAuth```.

Reply:
This is an optional plugin that is not included in the bundle by default. This is here for legacy reason. If we want to remove it, we should put a deprecation notice first.

Bitbucket's API specificaiton is troublesome enough, that introduces breaking changes frequently. I just don't want to add more to it from this library.

Eitherway, this removing this is not related to this PR at all. "If" we want to do this, this should be done in a separate PR.

Reply:
Ok, sounds reasonable!